File: server/src/dsa/Queue.java
package dsa;

import java.util.ArrayList;

public class Queue {
    private ArrayList<Object> data;
    public Queue() {
        data = new ArrayList<>();
    }

    public boolean isEmpty() {
        return data.isEmpty();
    }

    public void enqueue(Object item) {
        data.add(item);
    }

    public Object dequeue() {
        if (isEmpty()) return null;
        Object returnValue = data.get(0);
        data.remove(0);
        return returnValue;
    }
}

File: client/src/modules/dsa/Queue.js
let Queue = cc.Class.extend({

    ctor: function () {
        this.data = [];
    },

    isEmpty: function() {
        return this.data.length === 0;
    },

    enqueue: function (item) {
        this.data.push(item);
        return true;
    },

    dequeue: function () {
        if (this.isEmpty()) return undefined;
        return this.data.shift();
    }

})